### PR TITLE
Moved code into directory structure

### PR DESCRIPTION
**Pull request by telegraphic**
Creating pull request because I forgot I did this, better to discuss before our branches diverge too much!

### CODE REVIEW

**Comment by jeenriquez** [comment]
Great Danny! yes, I think we may need to chat about this. I see there are multiple things being addressed, some easier to merge than others. Could it be possible to separate/organize them?

**Comment by telegraphic** [comment]
Errm, maybe ask @david-macmahon? Most likely possible with git-fu, but beyond my general use knowledge.

**Comment by david-macmahon** [comment]
@jeenriquez Is there anything in particular you'd prefer NOT to merge?  It's probably possible to spend some time and split @telegraphic's commits into "sub-branches" and then merge them all back together, but if we're going to end up accepting all of the work anyway then that's probably not a worthwhile exercise.  OTOH, if there are parts you'd prefer to exclude or merge at a later time, then we can look into separating out those changes and merging the rest.

**Comment by telegraphic** [comment]
Bumping this so I can start on JOSS revisions.

To give more of an overview, I've created subdirectories `ephemeris`, `io`, and `plotting`. This is driven by the 'separation of concerns' and 'keep files small' mantras, and should make maintenance and extending much easier. Functionality should be _identical_ to before.

BEFORE:
```
blimpy
  |- calib_utils
  |- deprecated
```

AFTER
```
blimpy
  |- calib_utils
  |- deprecated
  |- plotting
  |- ephemeris
  |- io
```

* The `io` directory contains `file_wrapper.py`, the main file handler, `fil_writer.py` and `hdf_writer.py` for writing files, and `sigproc.py` for handling sigproc filterbank files. Anyone wanting to add a new reader or writer need only touch code in this directory.

* The `ephemeris` directory contains `compute_lst.py` for computing sidereal time, and `compute_lsrk.py`, which computes the LSRK for doppler correction of Earth rotation. The lat/lon/elevation of the Parkes and GBT telescopes is included in `config.py` here, so that other telescopes can be added straightforwardly in the future. 

* The `plotting` directory houses all the methods for plotting, that were previously methods of the `Filterbank`/`Waterfall` classes. Importantly, these can still be accessed from the `Waterfall` class via e.g. `waterfall.plot_spectrum()` as before (they are attached to the class using a `create_bound_method` function). The motivation here is to allow a) different plotting libraries to be added (e.g. [plotly](https://plot.ly/python)), and b) to allow non-blimpy use of the plotting functions by API-consistent code (which honestly is unlikely in the short term).

It's also easier to see which parts of the code are less tested by looking at the coverage report. Coverage increases to 68%:
```
Name                                                       Stmts   Miss  Cover
------------------------------------------------------------------------------
/data/bl/blimpy/blimpy/__init__.py                            25      8    68%
/data/bl/blimpy/blimpy/dice.py                               103     48    53%
/data/bl/blimpy/blimpy/ephemeris/__init__.py                   2      0   100%
/data/bl/blimpy/blimpy/ephemeris/compute_lsrk.py              28      0   100%
/data/bl/blimpy/blimpy/ephemeris/compute_lst.py               15      4    73%
/data/bl/blimpy/blimpy/ephemeris/config.py                     9      2    78%
/data/bl/blimpy/blimpy/fil2h5.py                              42     21    50%
/data/bl/blimpy/blimpy/gup2hdf.py                             61     50    18%
/data/bl/blimpy/blimpy/guppi.py                              272    135    50%
/data/bl/blimpy/blimpy/h52fil.py                              42     21    50%
/data/bl/blimpy/blimpy/io/__init__.py                          2      0   100%
/data/bl/blimpy/blimpy/io/fil_writer.py                       41      8    80%
/data/bl/blimpy/blimpy/io/file_wrapper.py                    396     99    75%
/data/bl/blimpy/blimpy/io/hdf_writer.py                       87     23    74%
/data/bl/blimpy/blimpy/io/sigproc.py                         157     28    82%
/data/bl/blimpy/blimpy/match_fils.py                          74     61    18%
/data/bl/blimpy/blimpy/plotting/__init__.py                    7      0   100%
/data/bl/blimpy/blimpy/plotting/config.py                     11      3    73%
/data/bl/blimpy/blimpy/plotting/plot_all.py                   69      5    93%
/data/bl/blimpy/blimpy/plotting/plot_kurtosis.py              16      2    88%
/data/bl/blimpy/blimpy/plotting/plot_spectrum.py              39      6    85%
/data/bl/blimpy/blimpy/plotting/plot_spectrum_min_max.py      44      5    89%
/data/bl/blimpy/blimpy/plotting/plot_time_series.py           26      3    88%
/data/bl/blimpy/blimpy/plotting/plot_utils.py                 10      1    90%
/data/bl/blimpy/blimpy/plotting/plot_waterfall.py             28      4    86%
/data/bl/blimpy/blimpy/utils.py                               60      0   100%
/data/bl/blimpy/blimpy/waterfall.py                          244     70    71%
------------------------------------------------------------------------------
TOTAL                                                       1910    607    68%
```

**Comment by telegraphic** [comment]
Main changes were 1) moving gup2hdf to deprecated (nobody uses it and it's not well tested), and adding voyager as an example to a new `examples/` directory.

```
Name                                                       Stmts   Miss  Cover
------------------------------------------------------------------------------
/data/bl/blimpy/blimpy/__init__.py                            25      8    68%
/data/bl/blimpy/blimpy/dice.py                               103     48    53%
/data/bl/blimpy/blimpy/ephemeris/__init__.py                   2      0   100%
/data/bl/blimpy/blimpy/ephemeris/compute_lsrk.py              28      0   100%
/data/bl/blimpy/blimpy/ephemeris/compute_lst.py               15      4    73%
/data/bl/blimpy/blimpy/ephemeris/config.py                     9      2    78%
/data/bl/blimpy/blimpy/fil2h5.py                              42     21    50%
/data/bl/blimpy/blimpy/guppi.py                              272    135    50%
/data/bl/blimpy/blimpy/h52fil.py                              42     21    50%
/data/bl/blimpy/blimpy/io/__init__.py                          2      0   100%
/data/bl/blimpy/blimpy/io/fil_writer.py                       41      8    80%
/data/bl/blimpy/blimpy/io/file_wrapper.py                    396     99    75%
/data/bl/blimpy/blimpy/io/hdf_writer.py                       87     23    74%
/data/bl/blimpy/blimpy/io/sigproc.py                         157     28    82%
/data/bl/blimpy/blimpy/match_fils.py                          74     61    18%
/data/bl/blimpy/blimpy/plotting/__init__.py                    7      0   100%
/data/bl/blimpy/blimpy/plotting/config.py                     11      3    73%
/data/bl/blimpy/blimpy/plotting/plot_all.py                   69      5    93%
/data/bl/blimpy/blimpy/plotting/plot_kurtosis.py              16      2    88%
/data/bl/blimpy/blimpy/plotting/plot_spectrum.py              39      6    85%
/data/bl/blimpy/blimpy/plotting/plot_spectrum_min_max.py      44      5    89%
/data/bl/blimpy/blimpy/plotting/plot_time_series.py           26      3    88%
/data/bl/blimpy/blimpy/plotting/plot_utils.py                 10      1    90%
/data/bl/blimpy/blimpy/plotting/plot_waterfall.py             28      4    86%
/data/bl/blimpy/blimpy/utils.py                               60      0   100%
/data/bl/blimpy/blimpy/waterfall.py                          244     70    71%
------------------------------------------------------------------------------
TOTAL                                                       1849    557    70%
```

**Comment by telegraphic** [comment]
The example is an ipynb, so renders in github. It's based on the tutorial from ages ago, see it here:
https://github.com/telegraphic/blimpy/blob/master/examples/voyager.ipynb

**Comment by telegraphic** [comment]
@FX196  would you mind having a look at these changes?

File: paper.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ authors:
     affiliation: "+, 1, 3"
   - name: Yuhong Chen
     affiliation: 2
```
Comment (FX196) [inline paper.md]:
```suggestion
     affiliation: 1    
```

**Comment by telegraphic** [comment]
@jeenriquez -- The JOSS reviewers are waiting on a response, so I'd like to merge this and then continue work on the [remaining issues](https://github.com/UCBerkeleySETI/blimpy/issues/101). My pref is to merge this branch then tackle any specific things to address

**Pull request merged**